### PR TITLE
feature:added permission configuration support for nacos 1.2

### PR DESCRIPTION
**Pull request by a364176773**
### Ⅰ. Describe what this PR did
added permission configuration support for nacos 1.2

### Ⅱ. Does this pull request fix one issue?

#2365

### CODE REVIEW

File: config/seata-config-nacos/src/main/java/io/seata/config/nacos/NacosConfiguration.java [hunk 1]
```diff
@@ -191,6 +194,17 @@ private static Properties getConfigProperties() {
             }
             properties.setProperty(PRO_NAMESPACE_KEY, namespace);
         }
+        if (StringUtils.isNotBlank(System.getProperty(USER_NAME))
```
Comment (objcoding) [inline config/seata-config-nacos/src/main/java/io/seata/config/nacos/NacosConfiguration.java]: Would it be better to write like this：
```java
String userName = StringUtils.isNotBlank(System.getProperty(USER_NAME))
            ? System.getProperty(USER_NAME)
            : FILE_CONFIG.getConfig(getNacosUserName());
        String password = StringUtils.isNotBlank(System.getProperty(PASSWORD))
            ? System.getProperty(PASSWORD)
            : FILE_CONFIG.getConfig(getNacosPassword());
        if (StringUtils.isNotBlank(userName) && StringUtils.isNotBlank(password)) {
            properties.setProperty(USER_NAME, userName);
            properties.setProperty(PASSWORD, password);
        }
```

**Comment by a364176773** [comment]
> Does registry conf also need permission?

@objcoding nacos1.2 auth only configuration centers are supported

File: config/seata-config-nacos/src/main/java/io/seata/config/nacos/NacosConfiguration.java (hunk 1 above)
Comment (a364176773) [reply config/seata-config-nacos/src/main/java/io/seata/config/nacos/NacosConfiguration.java]: > Would it be better to write like this：
> 
> ```java
> String userName = StringUtils.isNotBlank(System.getProperty(USER_NAME))
>             ? System.getProperty(USER_NAME)
>             : FILE_CONFIG.getConfig(getNacosUserName());
>         String password = StringUtils.isNotBlank(System.getProperty(PASSWORD))
>             ? System.getProperty(PASSWORD)
>             : FILE_CONFIG.getConfig(getNacosPassword());
>         if (StringUtils.isNotBlank(userName) && StringUtils.isNotBlank(password)) {
>             properties.setProperty(USER_NAME, userName);
>             properties.setProperty(PASSWORD, password);
>         }
> ```

thx,i under reference

**Comment by objcoding** [comment]
> > Does registry conf also need permission?
> 
> @objcoding nacos1.2 auth only configuration centers are supported

ok, now I know.

**Comment by a364176773** [comment]
> > > Does registry conf also need permission?
> > 
> > 
> > @objcoding nacos1.2 auth only configuration centers are supported
> 
> ok, now I know.

my fault, registry support

**Comment by a364176773** [comment]
> > > Does registry conf also need permission?
> > 
> > 
> > @objcoding nacos1.2 auth only configuration centers are supported
> 
> ok, now I know.

@objcoding @zjinlei @slievrly @xingfudeshi PTAL

**Comment by xingfudeshi** [comment]
LGTM.

**Comment by purgeyao** [comment]
```
  nacos {
    application = "seata-server"
    serverAddr = "10.1.1.97"
    namespace = "14ec3d04-0aab-4286-b7e1-8193f5dcb4ff"
    cluster = "default"
    username = "nacos"
    password = "nacos"
  }
```
配置了密码账号还是403 什么情况  还有其他配置么seata1.2.0

**Pull request merged**